### PR TITLE
Validate datatypes of filter values

### PR DESCRIPTION
**Pull request by hasegeli**

### CODE REVIEW

File: adminapi/base.py [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+    r'[0-9a-f]{1,4}:((:[0-9a-f]{1,4}){1,6})|'
+    r':((:[0-9a-f]{1,4}){1,7}|:)'
+    r')'
+)
+RE_32 = r'([0-9]|[1-2][0-9]|3[0-2])'
+RE_128 = r'([0-9]|[1-9][0-9]|1(1[0-9]|2[0-8]))'
+RE_MACADDR = r'([0-9a-f]{2}:){5}([0-9a-f]{2})'
+RE_DATE = r'[0-9]{1,4}-(0[0-9]|1[0-2])-([0-2][0-9]|3[0-1])'
+STR_BASED_DATATYPES = [
+    (IPv4Address, re_compile(r'\A' + RE_IPV4ADDR + r'\Z')),
```
Comment (HerrSpace) [inline adminapi/base.py]: I think this matches lines like '10.0.0.0/24'

File: adminapi/base.py [hunk 2]
```diff
@@ -388,3 +421,13 @@ def _validate_value(attribute_id, value, datatype=None):
...
     )
+
+
+def cast_datatype(value):
+    for datatype, regexp in STR_BASED_DATATYPES:
+        if regexp.match(str(value)):
+            # date constructor is special.
+            if datatype is date:
+                return date(*value.split('-', 2))
+            return datatype(value)
```
Comment (HerrSpace) [inline adminapi/base.py]: Which makes one of our tests crash here with an 
AddressValueError: Unexpected '/' in u'10.0.0.0/24'

Comment (hasegeli) [reply adminapi/base.py]: Fixed.

Comment (HerrSpace) [reply adminapi/base.py]: Confirmed!

**Pull request merged**